lib/did_you_mean/extra_features/key_error.rb

```diff
@@ -0,0 +1,25 @@
+module DidYouMean
+  module ExtraFeatures
+    module KeyErrorReciver
+      def fetch(name, *args, &block)
+        super
+      rescue KeyError => e
+        this = self
+        e.define_singleton_method(:receiver) { this }
+        e.define_singleton_method(:name) { name }
+        e.define_singleton_method(:args) { nil }
+        raise e
+      end
+    end
+    Hash.prepend KeyErrorReciver
+
+    class KeyNameChecker < MethodNameChecker
```

Comment:
I don't think it makes sense to let `KeyNameChecker ` inherit from `MethodNameChecker` since `KeyError` doesn't behave like `NameError`. It can just include `SpellCheckable` and implement what is needed.